wp-content/civi-extensions/goonjcustom/cli/assign-coordinating-poc.php

```diff
@@ -31,145 +32,168 @@ function getContactsFromGroup(): array {
   return $groupContacts->getIterator()->getArrayCopy();
 }
 
+/**
+ *
+ */
 function getStateOfficeId(int $stateId): ?int {
-    $offices = Contact::get(FALSE)
-      ->addSelect('id')
-      ->addWhere('contact_type', '=', 'Organization')
-      ->addWhere('contact_sub_type', 'CONTAINS', 'Goonj_Office')
-      ->addWhere('Goonj_Office_Details.Institution_Catchment', 'CONTAINS', $stateId)
-      ->execute();
-  
-    if (!$offices) {
-      echo "No office found for state ID $stateId\n";
-      return null;
-    }
-  
-    return $offices->first()['id'];
+  $offices = Contact::get(FALSE)
+    ->addSelect('id')
+    ->addWhere('contact_type', '=', 'Organization')
+    ->addWhere('contact_sub_type', 'CONTAINS', 'Goonj_Office')
+    ->addWhere('Goonj_Office_Details.Institution_Catchment', 'CONTAINS', $stateId)
+    ->execute()->first();
+
+  if ($offices) {
+    return $offices['id'];
+  }
+
+  echo "No office found for state ID $stateId. Using fallback coordinator.\n";
+
+  // Fetch the fallback coordinator group.
+  $fallbackGroups = Group::get(FALSE)
+    ->addWhere('Chapter_Contact_Group.Use_Case', '=', 'chapter-contacts')
```

Comment:
What is the reason for this where clause?

Reply:
In the chapter group use case, I have access to both chapter contacts and chapter team. but, I need to specifically check chapter contacts because it includes all contacts, whereas the chapter team only contains members of the Urban Ops team. That's why I'm verifying the chapter contacts in this case.